src/likelihoods/categorical.jl

```diff
@@ -36,6 +36,4 @@ function (l::CategoricalLikelihood)(f::AbstractVector{<:Real})
     return Categorical(l.invlink(f))
 end
 
-function (l::CategoricalLikelihood)(fs::AbstractVector)
-    return Product(Categorical.(l.invlink.(fs)))
-end
```

Comment:
is this a bug then @devmotion ?

Reply:
looks like this is assuming vectors of vectors (or vectors of tuples)?

Reply:
Could the following work then:
```julia
(l::AbstractLikelihood)(fs::AbstractVector) = Product(map(l, fs))
(l::CategoricalLikelihood)(f::AbstractVector{<:Real}) = Categorical(l.invlink(f))
```
?

Reply:
No, I don't think this is a bug. It is just the usual dispatch we also use in KernelFunctions - multiple samples form a vector, e.g., `ColVecs`, `RowVecs` or a vector of vectors. The single sample case is handled by the definition for `AbstractVector{<:Real}` above.